src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -64,44 +67,51 @@ public function process(ContainerBuilder $container)
 
         foreach ($container->findTaggedServiceIds($this->listenerTag, true) as $id => $events) {
             foreach ($events as $event) {
-                $priority = isset($event['priority']) ? $event['priority'] : 0;
-
-                if (!isset($event['event'])) {
-                    throw new InvalidArgumentException(sprintf('Service "%s" must define the "event" attribute on "%s" tags.', $id, $this->listenerTag));
+                $class = $container->getDefinition($id)->getClass();
+                if (!$r = $container->getReflectionClass($class)) {
```

Comment:
This call should be done only if the event attribute is not found: we should prevent as much as possible loading classes when not strictly required: this slows down compilation, thus DX.
I know Messenger passes violate this rule, that's an issue there that should be fixed (and not propagated :) )

Reply:
fixed. Code is a little bit less readable